### PR TITLE
EVA-3008 download assembly report before parsing

### PR DESCRIPTION
**Pull request by nitin-ebi**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/eva/contigalias/datasource/ENAAssemblyDataSource.java
Comment (sundarvenkata-EBI) [inline src/main/java/uk/ac/ebi/eva/contigalias/datasource/ENAAssemblyDataSource.java:72]: Optional: There is a discussion to be had as to whether we want this downloaded file should be permanent and if so, should we defer to the standard path where we currently download these assembly reports (via other automation)?

Comment (sundarvenkata-EBI) [reply src/main/java/uk/ac/ebi/eva/contigalias/datasource/ENAAssemblyDataSource.java:72]: **Optional:** Suggest refactoring the download code to a separate function

File: src/main/java/uk/ac/ebi/eva/contigalias/dus/PassiveAnonymousFTPClient.java
Comment (sundarvenkata-EBI) [inline src/main/java/uk/ac/ebi/eva/contigalias/dus/PassiveAnonymousFTPClient.java:106]: I don't know how reliable this method is... Is there a way to compare checksums instead?

**Review by sundarvenkata-EBI — APPROVED** [review]
Excellent work! Suggest doing a dry run with this code before merging.

File: src/main/java/uk/ac/ebi/eva/contigalias/datasource/ENAAssemblyDataSource.java
Comment (tcezard) [reply src/main/java/uk/ac/ebi/eva/contigalias/datasource/ENAAssemblyDataSource.java:72]: Would the webservice have access to NFS mounts ? 
I would not use the one we currently use because they have been sometime modified.

File: src/main/java/uk/ac/ebi/eva/contigalias/dus/PassiveAnonymousFTPClient.java
Comment (tcezard) [reply src/main/java/uk/ac/ebi/eva/contigalias/dus/PassiveAnonymousFTPClient.java:106]: In my experience File size is very reliable as long as the file is not tiny. 
I don't think we would have access to a file digest anyway.

File: src/main/java/uk/ac/ebi/eva/contigalias/datasource/ENAAssemblyDataSource.java
Comment (tcezard) [reply src/main/java/uk/ac/ebi/eva/contigalias/datasource/ENAAssemblyDataSource.java:72]: Actually I'm wondering if the files should be deleted once we're done processing it so that we're certain not to run our of space in the `/tmp` directory.

**Review by tcezard — APPROVED** [review]

File: src/main/java/uk/ac/ebi/eva/contigalias/entities/SequenceEntity.java
Comment (apriltuesday) [inline src/main/java/uk/ac/ebi/eva/contigalias/entities/SequenceEntity.java:49]: Probably just some context I missed, but why are we adding sequence length?  Is it just another useful property to have or is there another reason?

File: src/main/java/uk/ac/ebi/eva/contigalias/entities/ChromosomeId.java
Comment (apriltuesday) [inline src/main/java/uk/ac/ebi/eva/contigalias/entities/ChromosomeId.java:7]: Just for my understanding: this is type `String` because while an `AssemblyEntity` was annotated `@Id` in the main entity class, it's really using the id of `AssemblyEntity` which is `String insdcAccession`.... is that right?

**Review by apriltuesday — APPROVED** [review]
Tried to help with the great `genbank` -> `insdcAccession` name change, hope I didn't screw up...  Anyway, looks great!

File: pom.xml
Comment (nitin-ebi) [reply pom.xml:143]: added

File: src/main/java/uk/ac/ebi/eva/contigalias/datasource/ENAAssemblyDataSource.java
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/contigalias/datasource/ENAAssemblyDataSource.java]: added the code to delete file after being read

File: src/main/java/uk/ac/ebi/eva/contigalias/controller/admin/AdminController.java
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/contigalias/controller/admin/AdminController.java]: done

File: src/main/java/uk/ac/ebi/eva/contigalias/repo/ChromosomeRepository.java
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/contigalias/repo/ChromosomeRepository.java]: done

File: src/main/java/uk/ac/ebi/eva/contigalias/service/ChromosomeService.java
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/contigalias/service/ChromosomeService.java]: done

File: src/main/java/uk/ac/ebi/eva/contigalias/service/ChromosomeService.java
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/contigalias/service/ChromosomeService.java]: done

File: src/main/java/uk/ac/ebi/eva/contigalias/service/ChromosomeService.java
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/contigalias/service/ChromosomeService.java]: done

File: src/main/java/uk/ac/ebi/eva/contigalias/service/ChromosomeService.java
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/contigalias/service/ChromosomeService.java]: done

File: src/main/java/uk/ac/ebi/eva/contigalias/service/ChromosomeService.java
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/contigalias/service/ChromosomeService.java]: done

File: src/main/java/uk/ac/ebi/eva/contigalias/entities/SequenceEntity.java
Comment (nitin-ebi) [reply src/main/java/uk/ac/ebi/eva/contigalias/entities/SequenceEntity.java:49]: Suggested by Tim to include as he generally uses it for comparing

File: src/test/java/uk/ac/ebi/eva/contigalias/controller/contigalias/ContigAliasHandlerTest.java
Comment (nitin-ebi) [reply src/test/java/uk/ac/ebi/eva/contigalias/controller/contigalias/ContigAliasHandlerTest.java:446]: added test for retying

**Review by sundarvenkata-EBI — APPROVED** [review]

**Review by apriltuesday — APPROVED** [review]

**Pull request merged**
